### PR TITLE
[castai-sec-agent] Add castai-sec-agent charts

### PR DESCRIPTION
**Pull request by damejeras**

### CODE REVIEW

File: charts/castai-sec-agent/templates/secret.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{{- if or .Values.apiKey .Values.castai.apiKey }}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: castai-sec-agent
+  namespace: {{ .Values.namespace }}
```
Comment (matas-cast) [inline charts/castai-sec-agent/templates/secret.yaml:7]: I think namespace should be removed (Helm will use .Release.Namespace)

File: charts/castai-sec-agent/templates/rbac.yaml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    verbs: [ "delete" ]
+---
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: castai-sec-agent
+  labels:
+    {{- include "castai-sec-agent.labels" . | nindent 4 }}
+rules:
+  # Agent has full access in castai-sec-agent namespace. This is required to fully manage castai components.
```
Comment (matas-cast) [inline charts/castai-sec-agent/templates/rbac.yaml:38]: Comment probably not relevant for now

File: charts/castai-sec-agent/ci/test-values.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+resources:
+  requests:
+    cpu: 100m
+    memory: 64Mi
+  limits:
+    cpu: 1000m
+    memory: 256Mi
+
+podLabels:
+  team: core
```
Comment (matas-cast) [inline charts/castai-sec-agent/ci/test-values.yaml:13]: team: security
🙏

File: charts/castai-sec-agent/values.yaml [hunk 4]
```diff
@@ -0,0 +1,102 @@
+# Default values for castai-sec-agent.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+replicas: 2
```
Comment (matas-cast) [inline charts/castai-sec-agent/values.yaml:5]: I think we talked in short chat in the morning, replica 1 + leader election disabled

File: charts/castai-sec-agent/values.yaml [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+securityContext: {}
+
+serviceAccount:
+  create: true
+  # Annotations to add to the service account.
+  annotations: {}
+  name: "castai-sec-agent"
+
+# Settings for configuring k8s client used in castai-sec-agent.
+k8sApiClient:
```
Comment (matas-cast) [inline charts/castai-sec-agent/values.yaml:97]: I think we won't need this (for now)

Comment (damejeras) [reply charts/castai-sec-agent/values.yaml:97]: this is for ratelimiting requests to kube api. Im sure we will need this when we will start creating containers with our tools.

**Review by matas-cast — APPROVED** [review]

File: charts/castai-sec-agent/templates/rbac.yaml [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  name: castai-sec-agent
+  labels:
+    {{- include "castai-sec-agent.labels" . | nindent 4 }}
+rules:
+  # For installing/updating CAST AI components.
+  - apiGroups: [ "rbac.authorization.k8s.io" ]
+    resources: [ "roles", "clusterroles", "clusterrolebindings" ]
+    resourceNames: [ "castai-sec-agent" ]
+    verbs: [ "get", "patch", "update", "delete", "escalate" ]
+  - apiGroups: [ "" ]
```
Comment (anjmao) [inline charts/castai-sec-agent/templates/rbac.yaml:26]: We don't need delete namespace permissions.

**Review by anjmao — APPROVED** [review]

**Pull request merged**